test-dev/util.c

```diff
@@ -507,7 +507,7 @@ void dump_module(struct xmp_module *mod, FILE *f)
 			MD5Update(&ctx, xxs->data, len);
 		} else if (len > 0) {
 			/* The data file for fracture.stm had this hash for empty samples... */
-			MD5Update(&ctx, "1", 1);
+			MD5Update(&ctx, (unsigned char *)"1", 1);
```

Comment:
This is ugly.  No other choice I guess?

Reply:
I don't see any other obvious way to address that. gcc doesn't care about it, it's just clang but it's good to keep the compilation with no warnings. I think we should test with Visual C as well (and do it in Travis if possible).

Reply:
I'm not even sure this is the correct thing to do here, this is just what I had to add to reproduce the hashes in the fracture.stm data file. It's not clear to me how the Python version of this util even came up with this hash (maybe it's hashing a string representation of the length when no data is available?).

Reply:
Something like this should do:
```
diff --git a/test-dev/util.c b/test-dev/util.c
index b9bbba5..dd6a1a2 100644
--- a/test-dev/util.c
+++ b/test-dev/util.c
@@ -507,7 +507,8 @@ void dump_module(struct xmp_module *mod, FILE *f)
 			MD5Update(&ctx, xxs->data, len);
 		} else if (len > 0) {
 			/* The data file for fracture.stm had this hash for empty samples... */
-			MD5Update(&ctx, (unsigned char *)"1", 1);
+			static const unsigned char one_str[] = {'1',0};
+			MD5Update(&ctx, one_str, 1);
 		}
 		MD5Final(d, &ctx);
 		fprintf(f, " ");
```